src/pages/api/oauth/callback.ts

```diff
@@ -1,32 +1,48 @@
-import { HttpStatus } from "@fwl/web";
+import { Endpoint, HttpStatus } from "@fwl/web";
+import {
+  errorMiddleware,
+  loggingMiddleware,
+  recoveryMiddleware,
+  tracingMiddleware,
+} from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
 
 import { Handler } from "@src/@types/core/Handler";
 import { getAndPutUser } from "@src/commands/get-and-put-user";
 import { oauth2Client } from "@src/config";
-import { withLogger } from "@src/middlewares/with-logger";
+import { logger } from "@src/logger";
 import { withSentry } from "@src/middlewares/with-sentry";
 import { wrapMiddlewares } from "@src/middlewares/wrapper";
+import getTracer from "@src/tracer";
 import { setServerSideCookies } from "@src/utils/server-side-cookies";
 import { decryptVerifyAccessToken } from "@src/workflows/decrypt-verify-access-token";
 
-const handler: Handler = async (request, response): Promise<void> => {
-  if (request.method === "GET") {
+const tracer = getTracer();
+
+const handler: Handler = (request, response): Promise<void> => {
+  return tracer.span("callback handler", async (span) => {
     const {
       access_token,
       refresh_token,
     } = await oauth2Client.getTokensFromAuthorizationCode(
       `${request.query.code}`,
     );
 
+    span.setDisclosedAttribute("`authorization_code`", request.query.code);
```

Comment:
```suggestion
    span.setAttribute("`authorization_code`", request.query.code);
```

Reply:
So, I thought about this, but since we get it from the query string, should we really hide it?
And if so, what the point of `span.setAttribute("`authorization_code`", request.query.code);` ?

Reply:
Well, it's a code with one usage so it's not totally secret but we never will look its value against Connect neither (let's be honest). So in my mind it's just to get into the habit of setting attribute without disclosing everything.

> And if so, what the point 
to distinguish calls that have been made with an `?code=` from those with `?error=`